### PR TITLE
Revamp account registration

### PR DESCRIPTION
**Pull request by diegoalzate**
# overview
- migrate Tanstack/form to react-hook-form (Account page and Registration page)
- add credentialsGroup user attributes which is an array of credentials that allows users to put in where they studied, field and level of education

## why did we have to add React.forwardRef?
react-hook-form uses refs in their register function and our inputs because they are functional components they do not have and instance and therefore do not have a ref. React.forwardRef fixes this. 

## how do we handle credentialsGroups?
As the DB can tell credentialsGroups is another string value in the key -> value structure that we have for userAttributes. However the structure is just a JSON.stringify of:

```
type CredentialsGroup = {
  credential: string;
  institution: string;
  field: string;
}[];
```

we parse the string when we get the value from the db and handle it like it is a normal array

this pr also fixes some queries having unexpected behaviour when logging out. before all queries on logout would just get assigned a null value, now they will just get deleted.

### CODE REVIEW

File: src/components/form/Select.tsx [hunk 1]
```diff
@@ -21,8 +22,8 @@ const StyledSelect = styled.select`
   }
 `;
 
-function Select({ ...props }: SelectProps) {
```
Comment (camilovegag) [inline src/components/form/Select.tsx:24]: Would it bring any value to keep using `function` for our components?

```ts
function Input(props, ref) {
  return <StyledInput ref={ref} {...props} />;
}

const ForwardedInput = React.forwardRef(Input);

export default ForwardedInput;
```

File: src/pages/Account.tsx [hunk 2]
```diff
@@ -120,159 +178,118 @@ function Account() {
...
+                  })}
+                  placeholder="Institution"
+                />
+                <Input
+                  {...register(`userAttributes.credentialsGroup.${i}.field` as const, {
+                    required: 'Field is required',
+                  })}
+                  placeholder="Field"
+                />
+                <Button variant="text" onClick={() => remove(i)}>
```
Comment (camilovegag) [inline src/pages/Account.tsx:246]: Is there a way to only show the remove button to the added fields? and maybe making it a trash bin icon will save us some space on mobile as it is a bit broken atm

Same with the add new credential button, I think it should only be shown after the user filled the inputs for credential, institution and field.

Comment (camilovegag) [reply src/pages/Account.tsx:246]: We might need a FlexRow that turns into Column at 600px (our breakpoint)

**Review by camilovegag — APPROVED** [review]
Lots of changes but after testing it seemed to work fine, no console warnings. 

Great!

**Review by MartinBenediktBusch — CHANGES_REQUESTED** [review]
1. When selecting a credential the Institution and Filed fields should contain an example. Otherwise, people will insert unstructured random stuff. Furthermore, what to include into Field is not entirely clear. I would do the following: 
- Institution (e.g. University of Copenhagen) 
- Field (e.g. Economics) 

For example, if someone went to the University of Bonn, but I only see institution I might insert something like "Educational Institution" instead of University of Bonn. However, with the example, I know what and howI should insert it. The same holds for field. Imagine   that I studied economics, but if there is just written field I might insert Microeconomics. However, this would be too specific but with the example a participant knows the level of generality that we are asking for. 

2.  What is JD? 

3.  When inserting "None" I should not be required to insert something in the institution and field "Field". Either it should not be required or it should be autofilled with None in that case. Otherwise, people will be prompted to insert random stuff. 

4. When selecting "Other" I do not have the ability to insert a credential but I am ask for an institution and field. So the response would look like (Other, University of Copenhagen, Economics). Either we should remove other or we should allow for a free text entry of other when selecting the credential.  

5. Same holds for the publication field. Provide an example URL in the field. For example, https://appliednetsci.springeropen.com/articles/10.1007/s41109-023-00588-x

6. Same holds for MEV url.

7. I would not ask for Publication or Contribution to MEV URls just like that. I would asked for the 3 to 5 most relevant publications/urls. I.e. insert up to 5 urls. We dont want someone with 100 publications to insert all urls.

File: src/components/form/Select.tsx (hunk 1 above)
Comment (diegoalzate) [reply src/components/form/Select.tsx:24]: Not sure

Comment (diegoalzate) [reply src/components/form/Select.tsx:24]: It seemed to cause console errors so i decided on not including this in this pr

**Review by MartinBenediktBusch — APPROVED** [review]

**Review by camilovegag — APPROVED** [review]

**Pull request merged**